src/node_data.js

```diff
@@ -47,7 +47,7 @@ function NodeData(nodeName, key) {
    * move within their parent.
    * @const
    */
-  this.key = key;
+  this.key = key ? key : null;
```

Comment:
:smile: That's actually what I normally would have written. There's not a ton of code in the repo for me to guess Google's code standards (that might be a good thing).